app/helpers/foreman_host_reports/reports_dashboard_helper.rb

```diff
@@ -0,0 +1,88 @@
+module ForemanHostReports
+  module ReportsDashboardHelper
+    def host_reports_searchable_links(name, search, counter, format)
+      content_tag :li do
+        content_tag(:span, raw('&nbsp;'), :class => 'label', :style => "background-color:" + host_reports_report_color[counter]) +
+          raw('&nbsp;') +
+          link_to(name, hosts_path(:search => search), :class => "dashboard-links") +
+          content_tag(:h4, totals(format)[counter])
+      end
+    end
+
+    def host_reports_report_color
+      {
+        :change_hosts => "#4572A7",
+        :failure_hosts => "#AA4643",
+        :nochange_hosts => "#DB843D",
+        :disabled_hosts => "#92A8CD",
+      }
+    end
+
+    def host_reports_get_overview(options = {})
+      format = options[:format]
+      state_labels = {
+        change_hosts: _('Change'),
+        failure_hosts: _('Failure'),
+        nochange_hosts: _('No Change'),
+        disabled_hosts: _('Disabled alerts'),
+      }
+      state_filters = {
+        change_hosts: report_changes(format),
+        nochange_hosts: report_nochanges(format),
+        failure_hosts: report_failures(format),
+        disabled_hosts: disabled_alerts,
+      }
+
+      {
+        data: state_labels.map { |key, label| [label, state_filters[key], host_reports_report_color[key]] },
+        searchUrl: hosts_path(search: '~VAL~'),
+        searchFilters: state_filters.each_with_object({}) do |(key, filter), filters|
+          filters[state_labels[key]] = filter
+        end,
+      }
+    end
+
+
+    def latest_reports
+      hosts = Host.authorized(:view_hosts, Host)
+      @latest_reports ||= HostReport.authorized(:view_host_reports).my_reports
+                                    .where(host_id: hosts.pluck(:id))
+                                    .reorder(:last_report => :desc)
+                                    .limit(9)
+                                    .preload(:host)
+    end
+
+    def latest_reports?
+      latest_reports.limit(1).present?
+    end
+
+    private
+
+
+    def totals(format)
+      @totals ||= {
+        :total_hosts => Host.all.count,
+        :change_hosts => report_changes(format),
+        :nochange_hosts => report_nochanges(format),
+        :failure_hosts => report_failures(format),
+        :disabled_hosts => disabled_alerts,
+      }
+    end
```

Comment:
We cannot memorize values with arguments, each call should yield to a different result. I suggest to drop all memorization for now, from all methods in this helper.